Automattic/wp-calypso
# Emails: Improve forward form for email comparison

#### Changes proposed in this Pull Request

This change requests is adding a completely new way of adding mail forwards from the Emails comparison page.
It also prepares the frontend to add multiple forwards at once, but this is blocked by backend work.
This code changes also adds new validation in emails management to forbid the creation of repeated mailboxes when multiple mailboxes creation is allowed.

In further tasks, it would be also a possibility to extend the original component `email-forwarding-add-new.jsx` to remove the `email-forwarding-add-new-compact.jsx` which is a modification to avoid regression issues on the forward management screen.

#### Testing instructions

![image](https://user-images.githubusercontent.com/5689927/142179240-6fd1f2d4-c613-459c-b051-dff2f5c5d1e2.png)

**Test 1**
1. Go to Upgrade -> Emails
2. Add a single email forward, click add. Check that the forward is added.
3. Verify that you don't see a warning at the top of the screen about an existing email forward.
4. Assert that you have been redirected to forward management screen.

**Test 2**
1. Go to Upgrade -> Emails
2. Play around with validations:

- Add a mailbox with invalid characters (like $ or %)
- Add an invalid destination mail (wrong characters, or wrong domain)

**Remarks**
In a near future there will be a new implementation, which is to add several forwards at once. The backend isn't prepared for this task, but the frontend is being prepared for this task.

Related to 1200182182542585-as-1200414925548929
